### PR TITLE
[vcpkg] Commonize pipelines build powershell and don't use binary caching in CI (rather than PR) builds

### PR DESCRIPTION
**Pull request by BillyONeal**
This change obsoletes #11502 by making full rebuilds a normal part of our engineering process. Our CI builds are rolling: so even though this is typically going to take ~6-10 hours of machine time per triplet we build right now, it will batch up additional merged changes over that 6-10 hours and start again.

Hopefully this will give us a better understanding of ports that are flaky over time.

### CODE REVIEW

**Comment by cbezault** [comment]
I don't see where the CI gets batched. I think that a CI run gets kicked off for every new check-in, no?

**Comment by BillyONeal** [comment]
@cbezault It's in the trigger settings in the web interface:

image

**Comment by BillyONeal** [comment]
@cbezault It's in the trigger settings in the web interface:

image

**Comment by cbezault** [comment]
I think that can live in the yaml instead of in the interface but no complaints from me then.

File: scripts/azure-pipelines/test-modified-ports.ps1 [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    $BuildReason = $null
+)
+
+if (-Not (Test-Path "triplets/$Triplet.cmake")) {
+    Write-Error 'Incorrect triplet, please supply a valid triplet.'
+}
+
+$env:VCPKG_DOWNLOADS = Join-Path $WorkingRoot 'downloads'
+$buildtreesRoot = Join-Path $WorkingRoot 'buildtrees'
+$installRoot = Join-Path $WorkingRoot 'install'
```
Comment (dan-shaw) [inline scripts/azure-pipelines/test-modified-ports.ps1:53]: Should the `installRoot` be named `installed`?

File: scripts/azure-pipelines/test-modified-ports.ps1 [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    $WorkingRoot,
+    [ValidateNotNullOrEmpty()]
+    $ArtifactStagingDirectory = '.',
+    [ValidateNotNullOrEmpty()]
+    $ArtifactsDirectory = '.',
+    $BuildReason = $null
+)
+
+if (-Not (Test-Path "triplets/$Triplet.cmake")) {
+    Write-Error 'Incorrect triplet, please supply a valid triplet.'
```
Comment (dan-shaw) [inline scripts/azure-pipelines/test-modified-ports.ps1:48]: Can we print out the name of the triplet?

**Review by dan-shaw — APPROVED** [review]
Some comments, otherwise LGTM

**Comment by BillyONeal** [comment]
> I think that can live in the yaml instead of in the interface but no complaints from me then.

We use this same .yaml from both the PR pipeline and the CI pipeline, and those have different trigger settings, so it can't be in the yaml.

**Pull request merged**